src/scroll-timeline-css-parser.js

```diff
@@ -136,24 +136,13 @@ export class StyleParser {
   }
 
   findPreviousSiblingOrAncestorMatchingSelector(target, selector) {
-    // Target self is eligible, use that
-    if (target.matches(selector)) return target;
-
-    // Walk previous siblings
-    let prevSibling = target.previousElementSibling;
-    while (prevSibling) {
-      if (prevSibling.matches(selector)) {
-        return prevSibling;
-      }
-      prevSibling = prevSibling.previousElementSibling;
-    }
-
-    // Move up to parent
-    if (target.parentElement) {
-      return this.findPreviousSiblingOrAncestorMatchingSelector(target.parentElement, selector);
+    let candidate = target;
```

Comment:
Optional nit: OK to leave the comments.  

// Target self ... 
// Walk previous sibling, then the parent.
// No match

are all useful comments and make the code easier to read.

Reply:
:)

Reply:
Addressed in 48b223c56dab2c31bd689e967c8d6a0a33b33224